Reworking failure messages

jnyrup opened the pull request:
Contents of this PR:

* Use `context` in failure messages. This fixes #707 
* Reword "Did not expect" to "Expected*not to"

dennisdoomen commented:
Oh crap. This PR is doing the opposite of what I'm doing with #588

dennisdoomen commented:
See my ongoing work in #705. The parts where you added `{context}` is fine, but I've decided to keep the `Did not expect` message structure.

jnyrup commented:
Damn... 
I thought I remembered a conclusion in another PR, but just found #620 and saw we/you didn't came to a final conclusion.
I'll take out the rewording parts, 

(Reminder to self why splitting work into logical commits is A Good Idea ©)

dennisdoomen commented:
Sorry man. I'm not used to having such active contributors like you. Slack should help settle any discussions as well.

jnyrup commented:
No worries, I'm grateful for all the experience with API design and git by contributing.

I've removed the rewording and added missing periods instead.
